Atmel/Winc15x0/GHIElectronics.TinyCLR.Drivers.Atmel.Winc15x0/Winc15x0Interface.cs

```diff
@@ -0,0 +1,166 @@
+
+        public int Available(int socket) => this.networkController.Provider.Available(socket);
+
+        public bool Poll(int socket, int microSeconds, SelectMode mode) => this.networkController.Provider.Poll(socket, microSeconds, mode);
+
+        public int Send(int socket, byte[] buffer, int offset, int count, SocketFlags flags) => this.networkController.Provider.Send(socket, buffer, offset, count, flags);
+
+        public int Receive(int socket, byte[] buffer, int offset, int count, SocketFlags flags) => this.networkController.Provider.Receive(socket, buffer, offset, count, flags);
+
+        public int SendTo(int socket, byte[] buffer, int offset, int count, SocketFlags flags, SocketAddress address) => this.networkController.Provider.SendTo(socket, buffer, offset, count, flags, address);
+
+        public int ReceiveFrom(int socket, byte[] buffer, int offset, int count, SocketFlags flags, ref SocketAddress address) => this.networkController.Provider.ReceiveFrom(socket, buffer, offset, count, flags, ref address);
+
+        public void GetRemoteAddress(int socket, out SocketAddress address) => this.networkController.Provider.GetRemoteAddress(socket, out address);
+
+        public void GetLocalAddress(int socket, out SocketAddress address) => this.networkController.Provider.GetLocalAddress(socket, out address);
+
+        public void GetOption(int socket, SocketOptionLevel optionLevel, SocketOptionName optionName, byte[] optionValue) => this.networkController.Provider.GetOption(socket, optionLevel, optionName, optionValue);
+
+        public void SetOption(int socket, SocketOptionLevel optionLevel, SocketOptionName optionName, byte[] optionValue) => this.networkController.Provider.SetOption(socket, optionLevel, optionName, optionValue);
+
+        public int AuthenticateAsClient(int socketHandle, string targetHost, X509Certificate rootCertificate, SslProtocols sslProtocols) => this.networkController.Provider.AuthenticateAsClient(socketHandle, targetHost, rootCertificate, sslProtocols);
+
+        public int AuthenticateAsServer(int socketHandle, X509Certificate certificate, SslProtocols sslProtocols) => this.networkController.Provider.AuthenticateAsServer(socketHandle, certificate, sslProtocols);
+
+        public int SecureRead(int handle, byte[] buffer, int offset, int count) => this.networkController.Provider.SecureRead(handle, buffer, offset, count);
+
+        public int SecureWrite(int handle, byte[] buffer, int offset, int count) => this.networkController.Provider.SecureWrite(handle, buffer, offset, count);
+
+        public void GetHostByName(string name, out string canonicalName, out SocketAddress[] addresses) => this.networkController.Provider.GetHostByName(name, out canonicalName, out addresses);
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        public extern bool TurnOn();
+
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private extern byte[] NativeScan(out int numAp);
```

Comment:
Don't do anything like scan, power save, etc for now. We just want firmware update. We will add these other features later.